### ISSUE TITLE
Improve messaging around scheduler configuration

### ISSUE THREAD

**Issue by tchaton** [PyTorchLightning/pytorch-lightning#11128]
### To Reproduce

Users seem to be confused on how to properly setup their schedulers and Lightning doesn't provide help on possibly typos or miss configuration.

Related to conversation: https://github.com/PyTorchLightning/pytorch-lightning/discussions/9601

This seems to work properly.

```py
def test_lr_monitor_one_cycle_lr(tmpdir):

    class TestModel(BoringModel):

        def configure_optimizers(self):
            optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
            lr_scheduler = optim.lr_scheduler.OneCycleLR(
                optimizer,
                max_lr=0.01,
                steps_per_epoch=64,
                epochs=10,
                three_phase=True
            )
            return [optimizer], [{'scheduler': lr_scheduler, 'interval': 'step'}]

    model = TestModel()
    lr_monitor = LearningRateMonitor(logging_interval="step")
    trainer = Trainer(
        default_root_dir=tmpdir,
        max_epochs=10,
        callbacks=[lr_monitor],
    )
    trainer.fit(model)
    print(f"tensorboard --logdir {tmpdir}")
```

<img width="448" alt="Screenshot 2021-12-17 at 11 02 26" src="https://user-images.githubusercontent.com/12861981/146526974-c935a3ac-3468-410a-984b-5fc554f2db03.png">

This won't have any impacts as the key used is `lr_scheduler` and not `scheduler`. Lightning should raise a warning or a MisConfiguration there.

```py
def test_lr_monitor_one_cycle_lr(tmpdir):

    class TestModel(BoringModel):

        def configure_optimizers(self):
            optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
            lr_scheduler = optim.lr_scheduler.OneCycleLR(
                optimizer,
                max_lr=0.01,
                steps_per_epoch=64,
                epochs=10,
                three_phase=True
            )
            return {'scheduler': lr_scheduler, 'interval': 'step', 'optimizer': optimizer}

    model = TestModel()
    lr_monitor = LearningRateMonitor(logging_interval="step")
    trainer = Trainer(
        default_root_dir=tmpdir,
        max_epochs=10,
        callbacks=[lr_monitor],
    )
    trainer.fit(model)
    print(f"tensorboard --logdir {tmpdir}")
```

<img width="452" alt="Screenshot 2021-12-17 at 11 06 12" src="https://user-images.githubusercontent.com/12861981/146527480-86161d17-1d0b-4b55-9c9a-8e1d80fe839b.png">

And if I use `lr_scheduler`, it seems to not provide the right curve as this is computed on epoch. However, Lightning doesn't validate extra keys are provided such as `interval`.

```py
def test_lr_monitor_one_cycle_lr(tmpdir):

    class TestModel(BoringModel):

        def configure_optimizers(self):
            optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
            lr_scheduler = optim.lr_scheduler.OneCycleLR(
                optimizer,
                max_lr=0.01,
                steps_per_epoch=64,
                epochs=10,
                three_phase=True
            )
            return {'lr_scheduler': lr_scheduler, 'interval': 'step', 'optimizer': optimizer}

    model = TestModel()
    lr_monitor = LearningRateMonitor(logging_interval="step")
    trainer = Trainer(
        default_root_dir=tmpdir,
        max_epochs=10,
        callbacks=[lr_monitor],
    )
    trainer.fit(model)
```

<img width="438" alt="Screenshot 2021-12-17 at 11 54 38" src="https://user-images.githubusercontent.com/12861981/146534013-f29a540c-cad3-4368-b01d-6b757f9e3bd6.png">

The right way should be:

```py
def test_lr_monitor_one_cycle_lr(tmpdir):

    class TestModel(BoringModel):

        def configure_optimizers(self):
            optimizer = torch.optim.SGD(model.parameters(), lr=0.1, momentum=0.9)
            lr_scheduler = optim.lr_scheduler.OneCycleLR(
                optimizer,
                max_lr=0.01,
                steps_per_epoch=64,
                epochs=10,
                three_phase=True
            )
            return {'lr_scheduler': {'scheduler': lr_scheduler, 'interval': 'step'}, 'optimizer': optimizer}

    model = TestModel()
    lr_monitor = LearningRateMonitor(logging_interval="step")
    trainer = Trainer(
        default_root_dir=tmpdir,
        max_epochs=10,
        callbacks=[lr_monitor],
    )
    trainer.fit(model)
    print(f"tensorboard --logdir {tmpdir}")
```

<img width="453" alt="Screenshot 2021-12-17 at 11 58 21" src="https://user-images.githubusercontent.com/12861981/146534500-db97fe54-481d-4cab-9461-cb69d96562df.png">

### Environment

- PyTorch Lightning Version (e.g., 1.5.0):
- PyTorch Version (e.g., 1.10):
- Python version (e.g., 3.9):
- OS (e.g., Linux):
- CUDA/cuDNN version:
- GPU models and configuration:
- How you installed PyTorch (`conda`, `pip`, source):
- If compiling from source, the output of `torch.__config__.show()`:
- Any other relevant information:

### Additional context

cc @tchaton
